### PR TITLE
Create tables initialization query

### PR DESCRIPTION
**Pull request by demarkok**
Проверьте на опечатки.
Попробуйте запустить на локальной базе данных. 
Это делается чем-то типа
`psql --host=localhost --username=postgres --dbname=sql_homework -f init.sql`
Попробуйте поделать `INSERT`.

### CODE REVIEW

File: init.sql [hunk 1]
```diff
@@ -0,0 +1,93 @@
+-- TODO: default values - think about it
+
+CREATE TYPE colour AS ENUM ('gold', 'silver', 'bronze');
+CREATE TYPE sex AS ENUM ('male', 'female');
+
+CREATE TABLE vehicles (
+    id          serial        PRIMARY KEY,
+    reg_number  text          NOT NULL,
```
Comment (Ivan-Veselov) [inline init.sql]: UNIQUE

File: init.sql [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    capacity    integer       CHECK (capacity BETWEEN 1 AND 500) NOT NULL
```
Comment (Ivan-Veselov) [inline init.sql]: Magic number 500

File: init.sql [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+);
+
+CREATE TABLE assignments (
+    id           serial   PRIMARY KEY,
+    date         date     NOT NULL,
+    time         time     NOT NULL,
```
Comment (Ivan-Veselov) [inline init.sql]: Replace pair of date-time fields with one timestamp field.

File: init.sql [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    description  text     NOT NULL,
+    vehicle_id   integer  REFERENCES vehicles (id)
```
Comment (Ivan-Veselov) [inline init.sql]: NULL

File: init.sql [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    time         time     NOT NULL,
+    description  text     NOT NULL,
+    vehicle_id   integer  REFERENCES vehicles (id) 
+);
+
+CREATE TABLE volunteers (
+    id                serial   PRIMARY KEY,
+    name              text     NOT NULL,
+    telephone_number  text     NOT NULL,
+    card_number       integer  NOT NULL
```
Comment (Ivan-Veselov) [inline init.sql]: UNIQUE

File: init.sql [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    telephone_number  text     NOT NULL,
+    card_number       integer  NOT NULL 
+);
+
+CREATE TABLE buildings (
+    id            serial   PRIMARY KEY,
+    street        text     NOT NULL,
+    house_number  integer  NOT NULL,
+    type          text     NOT NULL,
+    name          text
```
Comment (Ivan-Veselov) [inline init.sql]: May be it is better to explicitly declare attribute with NULL.

File: init.sql [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+CREATE TABLE athletes (
+    id               serial   PRIMARY KEY,
+    name             text     NOT NULL,
+    sex              sex      NOT NULL,
+    height           integer  CHECK (height BETWEEN 50 AND 300) NOT NULL,
+    weight           integer  CHECK (weight BETWEEN 10 AND 650) NOT NULL,
+    age              integer  CHECK (age BETWEEN 2 AND 130) NOT NULL, 
+    accomodation_id  integer  REFERENCES buildings (id) NOT NULL,
+    card_number      integer  NOT NULL,
```
Comment (Ivan-Veselov) [inline init.sql]: UNIQUE

File: init.sql [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    delegation_id     integer  REFERENCES delegations (id) NOT NULL
+);
+
+CREATE TABLE athletes (
+    id               serial   PRIMARY KEY,
+    name             text     NOT NULL,
+    sex              sex      NOT NULL,
+    height           integer  CHECK (height BETWEEN 50 AND 300) NOT NULL,
+    weight           integer  CHECK (weight BETWEEN 10 AND 650) NOT NULL,
+    age              integer  CHECK (age BETWEEN 2 AND 130) NOT NULL,
```
Comment (Ivan-Veselov) [inline init.sql]: Probably it's better just to leave them positive.

File: init.sql [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    age              integer  CHECK (age BETWEEN 2 AND 130) NOT NULL, 
+    accomodation_id  integer  REFERENCES buildings (id) NOT NULL,
+    card_number      integer  NOT NULL,
+    delegation_id    integer  REFERENCES delegations (id) NOT NULL,
+    volunteer_id     integer  REFERENCES volunteers (id) NOT NULL
+);
+
+CREATE TABLE competitions (
+	id           serial   PRIMARY KEY,
+	sport        text     NOT NULL,
```
Comment (Ivan-Veselov) [inline init.sql]: Tabs instead of spaces.

File: init.sql [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    date         date     NOT NULL,
+    time         time     NOT NULL,
```
Comment (Ivan-Veselov) [inline init.sql]: timestamp

File: init.sql [hunk 11]
```diff
@@ -0,0 +1,93 @@
+-- TODO: default values - think about it
+
+CREATE TYPE colour AS ENUM ('gold', 'silver', 'bronze');
```
Comment (Ivan-Veselov) [inline init.sql]: 'medal_colour' is more self-explanatory.

**Review by Ivan-Veselov — CHANGES_REQUESTED** [review]

File: init.sql [hunk 12]
```diff
@@ -1,35 +1,34 @@
...
-CREATE TYPE colour AS ENUM ('gold', 'silver', 'bronze');
+CREATE TYPE medal_colour AS ENUM ('gold', 'silver', 'bronze');
 CREATE TYPE sex AS ENUM ('male', 'female');
 
 CREATE TABLE vehicles (
     id          serial        PRIMARY KEY,
-    reg_number  text          NOT NULL,
-    capacity    integer       CHECK (capacity BETWEEN 1 AND 500) NOT NULL
+    reg_number  text          NOT NULL UNIQUE,
+    capacity    integer       CHECK (capacity > 1) NOT NULL
```
Comment (Ivan-Veselov) [inline init.sql]: One is also possible.

**Review by Ivan-Veselov — CHANGES_REQUESTED** [review]

**Review by Ivan-Veselov — APPROVED** [review]

File: init.sql [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+    delegation_id    INTEGER  REFERENCES delegations (id) NOT NULL,
+    volunteer_id     INTEGER  REFERENCES volunteers (id) NOT NULL
+);
+
+CREATE TABLE competitions (
+    id           SERIAL     PRIMARY KEY,
+    sport        TEXT       NOT NULL,
+    timestamp    TIMESTAMP  NOT NULL,
+    description  TEXT       NOT NULL,
+    site_id      INTEGER    REFERENCES buildings (id) NOT NULL
```
Comment (shavkunov) [inline init.sql:64]: for me "place" is more explicit than "site", but it's ok

**Review by shavkunov — APPROVED** [review]

File: init.sql (hunk 13 above)
Comment (Ivan-Veselov) [reply init.sql:64]: [Site](http://www.oxfordlearnersdictionaries.com/definition/english/site_1) is more precise.

**Review by Fedannie — APPROVED** [review]

**Pull request merged**